crypto/impl_dispatch_test.cc

```diff
@@ -243,6 +247,39 @@ TEST_F(ImplDispatchTest, SHA512) {
 }
 #endif // OPENSSL_AARCH64
 
+
+#if defined(OPENSSL_X86_64) && !defined(MY_ASSEMBLER_IS_TOO_OLD_512AVX) && \
+    defined(RSAZ_512_ENABLED)
+TEST_F(ImplDispatchTest, BN_mod_exp_mont_consttime_x2) {
+  AssertFunctionsHit(
+      {
+          {kFlag_RSAZ_mod_exp_avx512_x2,
+	   is_x86_64_ &&
+           !is_assembler_too_old_avx512 &&
+	   ifma_avx512},
+      },
+      [] {
+	uint64_t res1 = 0;
+	uint64_t base1 = 0;
+	uint64_t exp1 = 0;
+	uint64_t m1 = 0;
+	uint64_t rr1 = 0;
+	uint64_t k0_1 = 0;
+	uint64_t res2 = 0;
+	uint64_t base2 = 0;
+	uint64_t exp2 = 0;
+	uint64_t m2 = 0;
+	uint64_t rr2 = 0;
+	uint64_t k0_2 = 0;
+        int modlen = 0;
+
+        RSAZ_mod_exp_avx512_x2(&res1, &base1, &exp1, &m1, &rr1, k0_1,
```

Comment:
I think we should be calling `BN_mod_exp_mont_consttime_x2` in exponentiation.c to make sure this function gets called. It may be why the test is failing as follows: the function is called where the conditions in `flag.second` are false.

```
[ RUN      ] ImplDispatchTest.BN_mod_exp_mont_consttime_x2
../crypto/impl_dispatch_test.cc:105: Failure
Expected equality of these values:
  flag.second
    Which is: false
  BORINGSSL_function_hit[flag.first] == 1
    Which is: true
Google Test trace:
../crypto/impl_dispatch_test.cc:103: 8
```

Reply:
Of course you're right. I'm not sure what I was thinking here.

Reply:
I even named the test the right thing! But got mixed up while filling it in. Anyways, https://github.com/aws/aws-lc/pull/1273/commits/92b9e3fa434ce4c950057ebfb4b791a0b0087413 fixes this, and with that I think we're caught up.